TeamTracker/app/src/main/java/com/example/teamtracker/fragments/TimerFragment.java

```diff
@@ -0,0 +1,122 @@
+        super.onViewCreated(view, savedInstanceState);
+        chronometer= view.findViewById(R.id.chronometer);
+        toggleButton=view.findViewById(R.id.toggleButton);
+        toggleButton.setChecked(true);
+        toggleButton.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton compoundButton, boolean isRunning) {
+                System.out.println(isRunning);
+                if(isRunning){
+
+                    chronometer.stop();
+                    long duration = SystemClock.elapsedRealtime() - chronometer.getBase();
+                    System.out.println(duration);
+                    showAddTaskDialog(getContext(),duration);
+                    chronometer.setBase(SystemClock.elapsedRealtime());
+
+                }else {
+
+                    chronometer.setBase(SystemClock.elapsedRealtime());
+                    chronometer.start();
+
+                }
+
+            }
+        });
+    }
+
+    private void showAddTaskDialog(Context context,long duration) {
+        final EditText projectNameEditText = new EditText(context);
+        AlertDialog dialog = new AlertDialog.Builder(context)
+                .setTitle("Hey you've worked for "+formatDuration(duration)+" !!")
+                .setMessage("What were you doing?")
+                .setView(projectNameEditText)
+                .setPositiveButton("Add Task", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        Toast.makeText(context, "Task Created Successfully!", Toast.LENGTH_SHORT).show();
+                        Task task = new Task(String.valueOf(projectNameEditText.getText()),"");
+                        database = RoomDB.getInstance(getContext());
+                        database.taskDao().insert(task);
+                        Fragment fragment = new TaskViewFragment();
+                        FragmentManager fragmentManager = getActivity().getSupportFragmentManager();
+                        fragmentManager.beginTransaction().replace(R.id.fragment_container, fragment).commit();
+                    }
+                })
+                .setNegativeButton("Cancel", null)
+                .create();
+        dialog.show();
+    }
+    private String formatDuration(long duration){
+        String time = "";
+        duration/=1000; // converting millisecond to second
+        long hour = duration/(60*60); // calculating hour
+        duration%=(60*60); // remaining seconds
+        long minute = duration/60; // calculating minute
+        long sec =  duration%60; //calculating second
+        // making the time string
+        if(hour != 0) time+= hour + "h ";
+        if(minute != 0) time+= minute + "m ";
+        if(sec != 0) time+= sec + "s";
+        else if(hour == 0 && sec == 0) time+= sec + "s";
+        return time;
+    }
```

Comment:
give proper spaces to them.
like 
duration = duration / 1000;
duration = duration % (60*60)

you have to also remove this 1000, 60*60 from here. these are called here magic number. instead you will take const variable like SECONDS_IN_ONE_HOUR = 60*60 and use SECONDS_IN_ONE_HOUR here